src/HealthGPS.Datastore/download_file.cpp

```diff
@@ -0,0 +1,61 @@
+#include "download_file.h"
+#include "HealthGPS/program_dirs.h"
+
+#include <curlpp/Easy.hpp>
+#include <curlpp/Exception.hpp>
+#include <curlpp/Options.hpp>
+#include <curlpp/cURLpp.hpp>
+#include <fmt/format.h>
+
+#include <fstream>
+#include <random>
+#include <sstream>
+
+namespace {
+std::filesystem::path get_temporary_file_path(const std::string &file_prefix,
+                                              const std::string &file_extension) {
+    const auto tmp_dir = hgps::get_temporary_directory();
+    std::filesystem::create_directories(tmp_dir);
+
+    std::mt19937 prng(std::random_device{}());
+    std::uniform_int_distribution<unsigned> rand;
+    std::filesystem::path path;
+    while (true) {
+        std::stringstream ss;
+        ss << file_prefix << std::hex << rand(prng) << file_extension;
+        path = tmp_dir / ss.str();
+        if (!std::filesystem::exists(path)) {
+            return path;
+        }
+    }
+}
+} // anonymous namespace
+
+namespace hgps::data {
+std::filesystem::path download_file(const std::string &url,
+                                    const std::filesystem::path &download_path) {
+    std::ofstream ofs{download_path};
+    if (!ofs) {
+        throw std::runtime_error(fmt::format("Failed to create file {}", download_path.string()));
+    }
+
+    curlpp::Cleanup cleanup;
```

Comment:
What does this do?

Reply:
Cleans up curl's global resources when it goes out of scope. Uses this function: https://curl.se/libcurl/c/curl_global_cleanup.html

Reply:
It looks a bit weird but I guess that's the C++/RAII way!